bin/p2-dsctl/main.go

```diff
@@ -0,0 +1,237 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"strconv"
+
+	"gopkg.in/alecthomas/kingpin.v2"
+
+	"github.com/Sirupsen/logrus"
+	ds_fields "github.com/square/p2/pkg/ds/fields"
+	"github.com/square/p2/pkg/kp"
+	"github.com/square/p2/pkg/kp/dsstore"
+	"github.com/square/p2/pkg/kp/flags"
+	"github.com/square/p2/pkg/logging"
+	pc_fields "github.com/square/p2/pkg/pc/fields"
+	"github.com/square/p2/pkg/pods"
+	"github.com/square/p2/pkg/types"
+	"github.com/square/p2/pkg/util"
+	"k8s.io/kubernetes/pkg/labels"
+)
+
+const (
+	CmdCreate  = "create"
+	CmdGet     = "get"
+	CmdList    = "list"
+	CmdEnable  = "enable"
+	CmdDisable = "disable"
+	CmdDelete  = "delete"
+	CmdUpdate  = "update"
+)
+
+var (
+	cmdCreate       = kingpin.Command(CmdCreate, "Create a daemon set.")
+	createAZ        = cmdCreate.Flag("az", "The availability zone of the pod cluster").Required().String()
+	createManifest  = cmdCreate.Flag("manifest", "Complete manifest - must parse as JSON!").Required().String()
+	createMinHealth = cmdCreate.Flag("minhealth", "The minimum health of the daemon set").Required().String()
+	createName      = cmdCreate.Flag("name", "The cluster name (ie. staging, production)").Required().String()
+	createPodID     = cmdCreate.Flag("pod", "The pod ID on the daemon set").Required().String()
+
+	cmdGet = kingpin.Command(CmdGet, "Show a daemon set.")
+	getID  = cmdGet.Flag("id", "The uuid for the daemon set").Required().String()
+
+	cmdList = kingpin.Command(CmdList, "List daemon sets.")
+
+	cmdEnable = kingpin.Command(CmdEnable, "Enable daemon set.")
+	enableID  = cmdEnable.Flag("id", "The uuid for the daemon set").Required().String()
+
+	cmdDisable = kingpin.Command(CmdDisable, "Disable daemon set.")
+	disableID  = cmdDisable.Flag("id", "The uuid for the daemon set").Required().String()
+
+	cmdDelete = kingpin.Command(CmdDelete, "Delete daemon set.")
+	deleteID  = cmdDelete.Flag("id", "The uuid for the daemon set").Required().String()
+
+	cmdUpdate       = kingpin.Command(CmdUpdate, "Update a daemon set.")
+	updateAZ        = cmdUpdate.Flag("az", "The availability zone of the pod cluster").String()
+	updateID        = cmdUpdate.Flag("id", "The uuid for the daemon set").Required().String()
```

Comment:
i feel that this shouldn't be a flag but an arg, so I can do `dsctl update abcd-3813 --az uswest` for example